### ISSUE TITLE
First time I try to log into AD it says Unauthorized

### ISSUE THREAD

**Issue by tfiduccia** [rancher/rancher#9419]
**Rancher versions:** v1.6.6.-rc1

**Steps to Reproduce:**
1.  Enter all the tad.rancher.io info and check TLS
2.  Try to Authenticate (I get and "Unauthorized" error
3.  Don't change anything and click on Authenticate again (works fine)

**Results:**  The first time I click on Authenticate, it won't authorize but the second time (without changing anything) it works fine.

**Comment by tfiduccia** [comment]
Version - v1.6.6-rc2
Whatever we changed for edit seems to have fixed this issue.

Verified fixed

**Comment by mrajashree** [comment]
Reopening to debug this as per discussion with @cjellick 
This is not always reproducible. But while running cattle out of eclipse, I have seen the `Unauthorized` error, and also `Identity externalIdType is invalid` and `Auth-service returned 500` errors from cattle. They are thrown because of the value of the settings `api.auth.provider.configured` and `api.auth.external.provider.configured` being `null` and `false` at the time `ExternalServiceAuthProvider:isConfigured` is evaluated. UpdateConfig from auth-service sets them to `ldapconfig` and `true` respectively, but as per cattle logs these settings are unset and then set again. The call to change these values to null or false does not come from auth-service.

**Comment by sangeethah** [comment]
When testing with HA setup having 3 rancher-servers in cluster , I observed the same issue where when AD auth was enabled in get ```Identity externalIdType is invalid``` error.
If i don't any values and retry again , there is no error and setting AD auth succeeds.

Also I see mutliple config reloads happening on each of the  rancher-server host:

```
time="2017-07-25T18:33:10Z" level=info msg="newProvider ldap" 
time="2017-07-25T18:33:10Z" level=info msg="Reload config is already in process, skipping" 
time="2017-07-25T18:33:10Z" level=info msg="Reload config is already in process, skipping" 
time="2017-07-25T18:33:10Z" level=info msg=" Auth provider configured ldapconfig" 
time="2017-07-25T18:33:10Z" level=info msg="Reload config is already in process, skipping" 
time="2017-07-25T18:33:10Z" level=info msg="Reload config is already in process, skipping" 
time="2017-07-25T18:33:10Z" level=info msg="Reload config is already in process, skipping" 
time="2017-07-25T18:33:10Z" level=info msg="Reload config is already in process, skipping" 
time="2017-07-25T18:33:10Z" level=info msg=" Auth provider configured ldapconfig" 
time="2017-07-25T18:33:10Z" level=info msg="Calling reload" 
time="2017-07-25T18:33:10Z" level=info msg="Reload config is already in process, skipping" 
time="2017-07-25T18:33:10Z" level=info msg="Reload config is already in process, skipping" 
time="2017-07-25T18:33:10Z" level=info msg=" Auth provider configured ldapconfig" 
time="2017-07-25T18:33:10Z" level=info msg=" Auth provider configured ldapconfig" 
time="2017-07-25T18:33:10Z" level=info msg="Now generating Ldap token" 
time="2017-07-25T18:33:34Z" level=info msg=" Auth provider configured ldapconfig" 
```

**Comment by prachidamle** [comment]
@mrajashree @cjellick With respect to the above observation of multiple reloads happening, I feel we should be able to avoid this and alleviate the  DB lookups of configuration from auth-service multiple times. 

Currently AuthServiceLauncher sends a reload signal to the auth-service on a bunch of settings. During enabling access control, since a new config is saved all these setting values change in DB. Cattle sends a reload signal per setting, so auth-service ends up getting multiple reloads. 
While we are processing a reload, we skip any other reload coming in in parallel hence the message "Reload config is already in process, skipping".
But still few of them happen to get processed in quick succession.

We should improve this further by limiting the number of settings we track and alert on.

Reloading the auth-service when say "auth.service.update.timestamp" changes during UpdateConfig from auth-service should alleviate this.

**Comment by deniseschannon** [comment]
Available in v1.6.8-rc1

**Comment by sangeethah** [comment]
Tested with v1.6.8-rc1

Not able to reproduce the issue where the first time I try to log into AD when it is enabled for the first time in a fresh install of rancher server fails.

Also tried enabling AD in a fresh install of HA setup multiple times. AD log in always succeeds in the first attempt.
